src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -209,13 +204,26 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     request.Headers.TransferEncodingChunked = true;
                 }
 
+                if (request.Version.Major != 1)
+                {
+                    // TODO #23134: Support 2.0
+                    throw new PlatformNotSupportedException($"Only HTTP 1.0 supported -- request.Version was {request.Version}");
+                }
+                else if (request.Version.Minor == 0 && request.Headers.TransferEncodingChunked == true)
```

Comment:
This should be:
```C#
&& request.HasHeaders && request.Headers.TransferEncodingChunked == true
```
rather than just:
```C#
&& request.Headers.TransferEncodingChunked == true
```
The way things are set up right now, the latter will force some objects to be lazily initialized, and we want to avoid that.